packages/flare/python/cron_job_ingest_events.py

```diff
@@ -1,72 +1,207 @@
 import json
-import sys
 import os
+import sys
+import time
+
+from datetime import date
+from datetime import datetime
+from enum import Enum
+from typing import Any
 from typing import Optional
 
+
 sys.path.insert(0, os.path.join(os.path.dirname(__file__), "vendor"))
 import vendor.splunklib.client as client
 
 from flare import FlareAPI
 
+
 APP_NAME = "flare"
 HOST = "localhost"
 SPLUNK_PORT = 8089
 REALM = APP_NAME + "_realm"
-KV_COLLECTION_NAME = "feednext"
+KV_COLLECTION_NAME = "event_ingestion_collection"
+CRON_JOB_THRESHOLD_SINCE_LAST_FETCH = 10 * 60
+
 
+class PasswordKeys(Enum):
+    API_KEY = "api_key"
+    TENANT_ID = "tenant_id"
 
-def main():
+
+class CollectionKeys(Enum):
+    CURRENT_TENANT_ID = "current_tenant_id"
+    START_DATE = "start_date"
+    NEXT_TOKEN = "next_"
+    TIMESTAMP_LAST_FETCH = "timestamp_last_fetch"
+
+
+def main() -> None:
     try:
-        # Example using a token
         splunk_service = client.connect(
             host=HOST,
             port=SPLUNK_PORT,
             app=APP_NAME,
             token=sys.stdin.readline().strip(),
         )
     except Exception as e:
+        # Use logger
         print(str(e), file=sys.stderr)
         raise Exception(str(e))
 
     app: client.Application = splunk_service.apps[APP_NAME]
-    flare_api = FlareAPI(app=app)
+    create_collection(app=app)
 
-    from_ = get_from_value(app=app)
-    event_feed = flare_api.retrieve_feed(from_=from_)
-    set_from_value(app=app, next=event_feed["next"])
+    # To avoid cron jobs from doing the same work at the same time, exit new cron jobs if a cron job is already doing work
+    last_fetched_timestamp = get_last_fetched(app)
+    if (
+        last_fetched_timestamp
+        and int(datetime.now().timestamp()) - last_fetched_timestamp
+        < CRON_JOB_THRESHOLD_SINCE_LAST_FETCH
+    ):
+        pass
```

Comment:
pass?

Reply:
Yep good catch, I had switch my return to pass to test my code